spicy/toolchain/src/compiler/visitors/validator.cc

```diff
@@ -210,6 +210,12 @@ struct PreTransformVisitor : public hilti::visitor::PreOrder<void, PreTransformV
             error("'stop' can only be used inside a 'foreach' hook", p);
     }
 
+    void operator()(const hilti::ctor::Exception& e, position_t p) {
+        if ( auto x = e.value().tryAs<hilti::expression::Ctor>() )
+            if ( ! x->type().isA<type::String>() )
+                error("exceptions need to be a string", p);
+    }
+
```

Comment:
This should probably move into the HILTI validator: currently all exception arguments must be a single strings anyways.